components/kubeflow/launcher/src/launch_tf_job.py

```diff
@@ -41,65 +43,90 @@
+    insert_volume(content['spec']['tfReplicaSpecs']['PS']['template']['spec'])
+  else:
+    content['spec']['tfReplicaSpecs'].pop('PS')
+
+  if workers:
     content['spec']['tfReplicaSpecs']['Worker']['replicas'] = workers
     content['spec']['tfReplicaSpecs']['Worker']['template']['spec']['containers'][0]['image'] = trainer_image
     content['spec']['tfReplicaSpecs']['Worker']['template']['spec']['containers'][0]['command'] = command
+    insert_volume(content['spec']['tfReplicaSpecs']['Worker']['template']['spec'])
+  else:
+    content['spec']['tfReplicaSpecs'].pop('Worker')
+
+  if has_master:
     content['spec']['tfReplicaSpecs']['MASTER']['template']['spec']['containers'][0]['image'] = trainer_image
     content['spec']['tfReplicaSpecs']['MASTER']['template']['spec']['containers'][0]['command'] = command
+    insert_volume(content['spec']['tfReplicaSpecs']['MASTER']['template']['spec'])
   else:
-    # If no workers and pss set, default is 1.
-    master_spec = content['spec']['tfReplicaSpecs']['MASTER']
-    worker_spec = content['spec']['tfReplicaSpecs']['Worker']
-    ps_spec = content['spec']['tfReplicaSpecs']['PS']
-    master_spec['template']['spec']['containers'][0]['image'] = trainer_image
-    master_spec['template']['spec']['containers'][0]['command'] = command
-    worker_spec['template']['spec']['containers'][0]['image'] = trainer_image
-    worker_spec['template']['spec']['containers'][0]['command'] = command
-    ps_spec['template']['spec']['containers'][0]['image'] = trainer_image
-    ps_spec['template']['spec']['containers'][0]['command'] = command
+    content['spec']['tfReplicaSpecs'].pop('MASTER')
 
   return content
 
 def main(argv=None):
   parser = argparse.ArgumentParser(description='Kubeflow TFJob launcher')
   parser.add_argument('--container-image', type=str,
                       help='''Container image to run using KubeFlow TFJob. The command line should be added after --.''')
+  parser.add_argument('--has-master', type=strtobool, default=True)
   parser.add_argument('--workers', type=int, default=0)
   parser.add_argument('--pss', type=int, default=0)
-  parser.add_argument('--cluster', type=str,
-                      help='GKE cluster set up for kubeflow. If set, zone must be provided. ' +
-                           'If not set, assuming this runs in a GKE container and current ' +
-                           'cluster is used.')
-  parser.add_argument('--zone', type=str, help='zone of the kubeflow cluster.')
-  parser.add_argument('--kfversion', type=str,
-                      default='v1alpha2',
-                      help='The version of the deployed kubeflow. ' +
-                           'If not set, the default version is v1alpha2')
+  parser.add_argument('--tfjob-version', type=str,
+                      default='v1beta2',
+                      help='The version of the deployed tfjob.' +
+                           'If not set, the default namespace is v1beta2.')
```

Comment:
NIT: I guess there is typo here about **namespace**. It should have been **tfjob-version**